ncr/src/main/kotlin/com/hartwig/actin/personalization/ncr/interpretation/filter/MetastaticDiagnosisRecordFilter.kt

```diff
@@ -0,0 +1,79 @@
+package com.hartwig.actin.personalization.ncr.interpretation.filter
+
+import com.hartwig.actin.personalization.ncr.datamodel.NcrRecord
+
+class MetastaticDiagnosisRecordFilter(override val logFilteredRecords: Boolean) : RecordFilter {
+    internal fun hasOnlyOneMetastaticDetection(records: List<NcrRecord>): Boolean {
+        val metastaticRecords =
+            records.filter { it.identification.metaEpis == 1 || it.identification.metaEpis == 2 }
+        val hasOnlyOneMetastaticDetection = metastaticRecords.size <= 1
+        if (!hasOnlyOneMetastaticDetection) {
+            log("Multiple metastatic detections found in tumor ${records.tumorId()}")
+        }
+        return hasOnlyOneMetastaticDetection
+    }
+
+    internal fun hasEmptyMetastaticFieldIfDetectionNotPresent(records: List<NcrRecord>): Boolean {
+        val absentMetastaticRecords = records.filter { it.identification.metaEpis == 0 }
+        val hasEmptyMetastaticFieldIfDetectionNotPresent = absentMetastaticRecords.all { record ->
+            val hasEmptyMetastaticDiagnosis = record.metastaticDiagnosis.metaProg1 == null &&
+                    record.metastaticDiagnosis.metaProg2 == null &&
+                    record.metastaticDiagnosis.metaProg3 == null &&
+                    record.metastaticDiagnosis.metaProg4 == null &&
+                    record.metastaticDiagnosis.metaProg5 == null &&
+                    record.metastaticDiagnosis.metaProg6 == null &&
+                    record.metastaticDiagnosis.metaProg7 == null &&
+                    record.metastaticDiagnosis.metaProg8 == null &&
+                    record.metastaticDiagnosis.metaProg9 == null &&
+                    record.metastaticDiagnosis.metaProg10 == null
```

Comment:
`with` and a list could help here:
```suggestion
            val hasEmptyMetastaticDiagnosis = with(record.metastaticDiagnosis) {
                listOf(
                    metaProg1,
                    metaProg2,
                    metaProg3,
                    metaProg4,
                    metaProg5,
                    metaProg6,
                    metaProg7,
                    metaProg8,
                    metaProg9,
                    metaProg10
                ).all { it == null }
```